fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/CreateResourceServiceImpl.java

```diff
@@ -194,8 +195,8 @@ private void checkParent(final String txId, final PersistentStorageSession pSess
             final ResourceHeaders parent;
             try {
                 // Make sure the parent exists.
+                // TODO: object existence can be from the index, but we don't have interaction model. Should we add it?
```

Comment:
That seems like a reasonable optimization.

Reply:
https://jira.lyrasis.org/browse/FCREPO-3258